Documentation/ExtensionArchitecture/Extbase/Reference/Validation/Validators/_FileUploadController.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+use TYPO3\CMS\Extbase\Mvc\Controller\FileUploadConfiguration;
+use TYPO3\CMS\Extbase\Validation\Validator\MimeTypeValidator;
+
+class SomeController extends ActionController
+{
+    public function initializeCreateAction(): void
+    {
+        // As Validators can contain state do not inject them
+        $mimeTypeValidator = GeneralUtility::makeInstance(MimeTypeValidator::class);
+        $mimeTypeValidator->setOptions(['allowedMimeTypes' => ['image/jpeg']]);
+
+        $fileHandlingServiceConfiguration = $this->arguments->getArgument('myArgument')->getFileHandlingServiceConfiguration();
+        $fileHandlingServiceConfiguration->addFileUploadConfiguration(
+            (new FileUploadConfiguration('myPropertyName'))
+                ->setRequired()
+                ->addValidator($mimeTypeValidator)
+                ->setMaxFiles(1)
+                ->setUploadFolder('1:/user_upload/files/'),
+        );
+
+        $this->arguments->getArgument('myArgument')->getPropertyMappingConfiguration()->skipProperties('myPropertyName');
```

Comment:
I don't understand that part. Skipping a property means: do not map that file into your model. So, in model are all other data from form, but file is missing. Yeah. Maybe it's OK, as the FileUploadConfiguration handles the file upload. BUT: Ich you store your model with an empty file, won't it overwrite the just stored file with NULL again?!?

Reply:
I took the example from the changelog entry, so @garvinhicking would have to anser these questions

Reply:
If the property is skipped, no file is uploaded. Only if everything is "there" an uploaded file will be persisted. So no overwriting here.

Reply:
The same is done in `FileHandlingService->addUploadConfigurationForProperty()`. Since uploads and file reference mapping is not done via ProperyMapper, it is required to exclude file uploads fields from property mapper. 

I do not really get to part with "store your model with an empty file", because if the create action is resposible to handle and persist incoming data including file uploads, why would someone set the file field value to null (=no file) again in the same method? This would be inconsistent and result in orphaned files in the upload folder.

Reply:
@derhansen Hi. The default value of `myPropertyName` is declared as following:

```
protected ?FileReference $file = null;
```

So, if `myPropertyName` is skipped, the validator will be executed, the file will be uploaded, but as property mapping to `myPropertyName` is skipped the file reference identifier will not be set within the domain model. So, if you store your model, the `null` will be used and overwrites previous created file uploads. Not tested, but my understanding of that.

Reply:
@froemken The mapping of file properties is performed without the extbase property mapper. That is also the reason, why property mapping in general for `FileUpload` attribute/annotatied properties is skipped in core. The `null` value will actually only be set for initially persisted objects (e.g. in a `createAction`). As soon as a file  property of an object has a value (= FileReference), modifications to the object will be applied to the object, which is fetched from persistence (e.g. Object fetched by `PersistentObjectConverter`) and thereby the object will have previously saved file references set also if property mapping for a file field is skipped.